agent-lib/src/main/java/org/terracotta/angela/agent/AgentController.java

```diff
@@ -251,95 +243,49 @@ public boolean installVoter(InstanceId instanceId, TerracottaVoter terracottaVot
+      if (!dirs.isPresent()) {
         return false;
       }
 
-      logger.info("Installing kit on host {} from {}", hostName, distribution);
-      File kitDir = kitManager.installKit(license, Collections.singletonList(hostName));
-      File workingDir = kitManager.getWorkingDir().toFile();
       clusterToolInstalls.computeIfAbsent(instanceId, id -> {
         BiFunction<Map<String, String>, String[], ToolExecutionResult> operation = (env, command) -> {
           DistributionController distributionController = distribution.createDistributionController();
-          return distributionController.invokeClusterTool(kitDir, workingDir, securityRootDirectory, tcEnv, env, command);
+          return distributionController.invokeClusterTool(dirs.get().kitDir, dirs.get().workingDir, securityRootDirectory, tcEnv, env, command);
         };
-        return new ToolInstall(workingDir, operation);
+        return new ToolInstall(dirs.get().workingDir, operation);
       });
     }
     return true;
   }
 
   public boolean installConfigTool(InstanceId instanceId, String hostName, Distribution distribution,
                                    License license, String kitInstallationName, SecurityRootDirectory securityRootDirectory,
-                                   TerracottaCommandLineEnvironment tcEnv) {
+                                   TerracottaCommandLineEnvironment tcEnv, String kitInstallationPath) {
     ToolInstall configToolInstall = configToolInstalls.get(instanceId);
 
     if (configToolInstall == null) {
-      RemoteKitManager kitManager = new RemoteKitManager(instanceId, distribution, kitInstallationName);
-      if (!kitManager.isKitAvailable()) {
+
+      Optional<Dirs> dirs = Dirs.discover(instanceId, hostName, distribution, license, kitInstallationName, kitInstallationPath);
+      if (!dirs.isPresent()) {
         return false;
       }
 
-      logger.info("Installing kit for host {} from {}", hostName, distribution);
-      File kitDir = kitManager.installKit(license, Collections.singletonList(hostName));
-      File workingDir = kitManager.getWorkingDir().toFile();
       configToolInstalls.computeIfAbsent(instanceId, id -> {
         BiFunction<Map<String, String>, String[], ToolExecutionResult> operation = (env, command) -> {
           DistributionController distributionController = distribution.createDistributionController();
-          return distributionController.invokeConfigTool(kitDir, workingDir, securityRootDirectory, tcEnv, env, command);
+          return distributionController.invokeConfigTool(dirs.get().kitDir, dirs.get().workingDir, securityRootDirectory, tcEnv, env, command);
         };
-        return new ToolInstall(workingDir, operation);
+        return new ToolInstall(dirs.get().workingDir, operation);
       });
     }
     return true;
   }
 
-  private void enableTmsSecurity(File tmcProperties, TmsServerSecurityConfig tmsServerSecurityConfig) {
```

Comment:
Removing that is going to be a problem for tests downloading a kit since security will stop to working and you would need to update all existing tests.
We had a lot of (and that's a euphemism) discussions around what would be the responsibilities of the tests and what would be the responsibilities of Angela and having the Tms security config holding the security setup and let Angela do the necessary installation seemed to be the best decision.
I'm not against changing that if that's the consensus but be prepared to update several tests in different places.

Reply:
I will refactor... we need to keep that step if kit install dir is not supplied